packages/runtime/datastore/src/fluidHandle.ts

```diff
@@ -86,27 +60,13 @@ export class FluidObjectHandle<
 	 * {@inheritDoc @fluidframework/core-interfaces#IFluidHandle.attachGraph }
 	 */
 	public attachGraph(): void {
-		if (this.visible) {
-			return;
-		}
-
-		this.locallyVisible = true;
-		this.pendingHandlesToMakeVisible.forEach((handle) => {
-			handle.attachGraph();
-		});
-		this.pendingHandlesToMakeVisible.clear();
 		this.routeContext.attachGraph();
 	}
 
 	/**
 	 * {@inheritDoc @fluidframework/core-interfaces#IFluidHandle.bind}
 	 */
-	public bind(handle: IFluidHandleInternal) {
-		// If this handle is visible, attach the graph of the incoming handle as well.
-		if (this.visible) {
-			handle.attachGraph();
-			return;
-		}
-		this.pendingHandlesToMakeVisible.add(handle);
+	public bind(node: IAttachableNode): void {
+		this.routeContext.bind?.(node);
```

Comment:
This is not correct in today's world.  This DDS may not bound to the DataStore yet.  If it's not, we shouldn't have the DataStore binding the target handle.  If we get rid of the "detached DDS" modality (see AB#35138) then this is valid.